pkg/network/tracer/testutil/tcp.go

```diff
@@ -75,3 +75,16 @@ func (t *TCPServer) Shutdown() {
 		t.ln = nil
 	}
 }
+
+// GetFreePort returns a free port on localhost
+func GetFreePort() (port uint16, err error) {
+	var a *net.TCPAddr
+	if a, err = net.ResolveTCPAddr("tcp", "localhost:0"); err == nil {
+		var l *net.TCPListener
+		if l, err = net.ListenTCP("tcp", a); err == nil {
+			defer l.Close()
```

Comment:
Theoretically the port will be released here, but chances are it does not get re-allocated again for the duration of the test. If you want a 100% guarantee, you could just return the listener object instead.